classes/services/PKPSubmissionService.inc.php

```diff
@@ -296,6 +296,50 @@ public function getReviewAssignments($submission) {
 		return $reviewAssignmentDao->getBySubmissionId($submission->getId());
 	}
 
+	/**
+	 * Is this user allowed to view the author details?
+	 *
+	 * - Anyone can view published submission authors
+	 * - Reviewers can only view authors in open reviews
+	 * - Managers and admins can view authors of any submission
+	 * - Subeditors, authors and assistants can only view authors in assigned subs
+	 *
+	 * @param $user User
+	 * @param $submission Submission
+	 * @return boolean
+	 */
+	public function canUserViewAuthor($user, $submission) {
+
+		if ($submission->getDatePublished()) {
+			return true;
+		}
+
+		$reviewAssignments = $this->getReviewAssignments($submission);
+		foreach ($reviewAssignments as $reviewAssignment) {
+			if ($user->getId() == $reviewAssignment->getReviewerId()) {
+				return $reviewAssignment->getReviewMethod() == SUBMISSION_REVIEW_METHOD_OPEN ? true : false;
+			}
+		}
+
+		$contextId = $submission->getContextId();
+
+		if ($user->hasRole(array(ROLE_ID_MANAGER, ROLE_ID_SITE_ADMIN), $contextId)) {
```

Comment:
Because `ROLE_ID_SITE_ADMIN` has `context_id=0`, this will not correctly match site admins, I think.

Reply:
https://github.com/NateWr/pkp-lib/commit/baa69ab2a936ee406d0a272ef26c9f941fd2e97f

In my site, the admin has the journal manager role, and I'm unable to remove it or create a new admin user without it. Out of curiosity, is there a case where a site admin is not a manager? Do we ever need to check for admins when we're checking for managers?

Reply:
When a journal is created, the site admin doing the creating gets granted a manager role -- but this might be something that changes as we improve the SaaS capabilities of OJS, e.g. to accommodate service providers (admins) who don't have active roles in the journal at all. Plus if someone manually creates a second site admin account, they wouldn't have any roles in the journals.